src/DataProtection/DataProtection/src/Cng/GcmAuthenticatedEncryptor.cs

```diff
@@ -179,11 +180,25 @@ protected override byte[] DecryptImpl(byte* pbCiphertext, uint cbCiphertext, byt
                             dwFlags: 0);
                         UnsafeNativeMethods.ThrowExceptionForBCryptStatus(ntstatus);
                         CryptoUtil.Assert(cbDecryptedBytesWritten == cbPlaintext, "cbDecryptedBytesWritten == cbPlaintext");
-
-                        // At this point, retVal := { decryptedPayload }
-                        // And we're done!
-                        return retVal;
                     }
+#else
+                    ReadOnlySpan<byte> nonce, key, encrypted, tag;
+                    Span<byte> plaintext;
+                    unsafe
```

Comment:
Any reason not to do the Aes inside the `unsafe` block? You could use `vars` if you did that

Reply:
Nice, done